spyder/app/tests/test_mainwindow.py

```diff
@@ -316,26 +319,25 @@ def main_window(request, tmpdir):
         main_window.window = window
     else:
         window = main_window.window
-        # Close everything we can think of
-        window.editor.close_file()
-        window.projects.close_project()
-
-        if window.console.error_dialog:
-            window.console.close_error_dialog()
-
-        window.switcher.close()
-        for client in window.ipyconsole.get_clients():
-            window.ipyconsole.close_client(client=client, ask_recursive=False)
-        window.outlineexplorer.stop_symbol_services('python')
-        # Reset cwd
-        window.explorer.chdir(get_home_dir())
-        spyder_boilerplate = window.get_plugin(
-            'spyder_boilerplate', error=False)
-        if spyder_boilerplate is not None:
-            window.unregister_plugin(spyder_boilerplate)
 
     # Remove Kite (In case it was registered via setup.py)
     window.completions.providers.pop('kite', None)
+    # Wait until console is up
+    shell = window.ipyconsole.get_current_shellwidget()
+    qtbot.waitUntil(lambda: shell._prompt_html is not None,
+                    timeout=SHELL_TIMEOUT)
+    # _DummyThread are created if current_thread() is called from them
```

Comment:
```suggestion

    # _DummyThread are created if current_thread() is called from them.
```

For clarity as well.